### PR TITLE
[expo-document-picker][android][ios] Add mime type to response of getDocumentAsync()

### LINKED ISSUE #12764: Include File MIME Type on the response of getDocumentAsync()
### Summary

FIle Mime Type isn't included in the `getDocumentAsync() `Response

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android, iOS, Web

### SDK Version (managed workflow only)

_No response_

### Environment

Expo CLI 4.1.6 environment info:
    System:
      OS: macOS 10.15.6
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 12.18.0 - ~/.nvm/versions/node/v12.18.0/bin/node
      Yarn: 1.22.5 - /usr/local/bin/yarn
      npm: 7.8.0 - ~/.nvm/versions/node/v12.18.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.10.1 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.2, DriverKit 20.0, macOS 11.0, tvOS 14.2, watchOS 7.1
    IDEs:
      Android Studio: 4.0 AI-193.6911.18.40.6626763
      Xcode: 12.2/12B45b - /usr/bin/xcodebuild
    npmPackages:
      expo: ^39.0.0 => 39.0.3 
      react: 16.13.1 => 16.13.1 
      react-dom: 16.13.1 => 16.13.1 
      react-native: ~0.63.3 => 0.63.3 
      react-native-web: ~0.13.7 => 0.13.18 
    npmGlobalPackages:
      expo-cli: 4.1.6
    Expo Workflow: bare

### Reproducible demo or steps to reproduce from a blank project

Call getDocumentAsync from Expo Document Picker and console log the response. You will not see a file type there.

### PR DESCRIPTION
**Pull request by mstach60161**
# Why
Fixes #12764.
DocumentPicker.getDocumentAsync(options) didn't return mime type which can be useful to handle file correctly.

# How

- Added mimeType parameter on android.
- Added mimeType parameter on ios.
- Updated docs.

# Test Plan

- Native component list.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).

### CODE REVIEW

File: packages/expo-document-picker/android/src/main/java/expo/modules/documentpicker/DocumentPickerModule.kt [hunk 1]
```diff
@@ -20,6 +20,7 @@ import java.io.File
 import java.io.FileOutputStream
 import java.io.IOException
 
+
```
Comment (lukmccall) [inline packages/expo-document-picker/android/src/main/java/expo/modules/documentpicker/DocumentPickerModule.kt:23]:
```suggestion
```

File: packages/expo-document-picker/android/src/main/java/expo/modules/documentpicker/DocumentPickerModule.kt [hunk 2]
```diff
@@ -88,14 +89,19 @@ class DocumentPickerModule(
...
         val result = Bundle().apply {
           putString("type", "success")
           putString("uri", documentDetails.uri)
           putString("name", documentDetails.name)
+          putString("mimeType", documentDetails.mimeType)
+          if (documentDetails.mimeType == null) {
+            putParcelable("mimeType", null)
+          } else {
+            putString("mimeType", documentDetails.mimeType)
+          }
```
Comment (lukmccall) [inline packages/expo-document-picker/android/src/main/java/expo/modules/documentpicker/DocumentPickerModule.kt:104]:
```suggestion
		 documentDetails.mimeType?.let {
           putString("mimeType", documentDetails.mimeType)
		 }
```
You don't have to put null if the value isn't present. You can omit such value.

File: packages/expo-document-picker/ios/EXDocumentPicker/EXDocumentPickerModule.m [hunk 3]
```diff
@@ -8,6 +8,7 @@
 #import <UIKit/UIKit.h>
 #import <MobileCoreServices/MobileCoreServices.h>
 
+
```
Comment (lukmccall) [inline packages/expo-document-picker/ios/EXDocumentPicker/EXDocumentPickerModule.m:11]:
```suggestion
```

File: packages/expo-document-picker/src/types.ts [hunk 4]
```diff
@@ -11,6 +11,7 @@ export type DocumentResult =
       name: string;
       size: number;
       uri: string;
+      mimeType: string;
```
Comment (lukmccall) [inline packages/expo-document-picker/src/types.ts:14]:
```suggestion
      mimeType?: string;
```

**Review by lukmccall — CHANGES_REQUESTED** [review]
Looks good ✅ 
I've left some comments 😉

File: packages/expo-document-picker/CHANGELOG.md [hunk 5]
```diff
@@ -10,6 +10,8 @@
 
 - Added `AndroidManifest.xml` queries for intent handling. ([#13388](https://github.com/expo/expo/pull/13388) by [@EvanBacon](https://github.com/EvanBacon))
 
+- Added mimetype to getDocumentAsync result. ([#13702](https://github.com/expo/expo/pull/13702) by [@mstach60161](https://github.com/mstach60161))
+
```
Comment (bbarthec) [inline packages/expo-document-picker/CHANGELOG.md:14]:
```suggestion
- Added mimetype to getDocumentAsync result. ([#13702](https://github.com/expo/expo/pull/13702) by [@mstach60161](https://github.com/mstach60161))

```

**Review by bbarthec — COMMENTED** [review]
Looks good to me 👍

**Review by bbarthec — APPROVED** [review]
Looks good! 👏 
I'm landing this one! 🥳

**Pull request merged**